s3-encryption/client.md

```diff
@@ -0,0 +1,142 @@
+When disabled, the S3EC MUST NOT decrypt objects encrypted using legacy wrapping algorithms; it MUST throw an exception when attempting to decrypt an object encrypted with a legacy wrapping algorithm.
+
+#### Enable Legacy Unauthenticated Modes
+
+The S3EC MUST support the option to enable or disable legacy unauthenticated modes (content encryption algorithms).
+The option to enable legacy unauthenticated modes MUST be set to false by default.
+When enabled, the S3EC MUST be able to decrypt objects encrypted with all content encryption algorithms (both legacy and fully supported).
+When disabled, the S3EC MUST NOT decrypt objects encrypted using legacy content encryption algorithms; it MUST throw an exception when attempting to decrypt an object encrypted with a legacy content encryption algorithm.
+
+#### Enable Delayed Authentication
+
+The S3EC MUST support the option to enable or disable Delayed Authentication mode.
+Delayed Authentication mode MUST be set to false by default.
+When enabled, the S3EC MAY release plaintext from a stream which has not been authenticated.
+When disabled the S3EC MUST NOT release plaintext from a stream which has not been authenticated.
+
+#### Set Buffer Size
+
+The S3EC SHOULD accept a configurable buffer size which refers to the maximum ciphertext length to store in memory when delayed authentication mode is disabled.
+
+### Wrapped S3 Client(s)
+
+The S3EC MUST support the option to provide an SDK S3 client instance during its initialization.
+The S3EC MUST NOT support use of S3EC as the provided S3 client during its initialization; it MUST throw an exception in this case.
+
+### Instruction File Configuration
+
+The S3EC MAY support the option to provide Instruction File Configuration during its initialization.
+If the S3EC in a given language supports Instruction Files, then it MUST accept Instruction File Configuration during its initialization.
+In this case, the Instruction File Configuration SHOULD be optional, such that its default configuration is used when none is provided.
+
+### Inherited SDK Configuration
+
+The S3EC MAY support directly configuring the wrapped SDK clients through its initialization.
+For example, the S3EC MAY accept a credentials provider instance during its initialization.
+If the S3EC accepts SDK client configuration, the configuration MUST be applied to all wrapped S3 clients.
+If the S3EC accepts SDK client configuration, the configuration MUST be applied to all wrapped SDK clients including the KMS client.
+If the S3EC accepts any SDK client configuration options, then the S3EC should support all possible configuration options.
+
+### Other Dependencies
+
+#### Randomness
+
+The S3EC MAY accept a source of randomness during client initialization.
+The inclusion of a source of randomness is subject to language availability.
+
+## API Operations
```

Comment:
for readability i would separate this in two subheading sections
- MUST support:
  - [GetObject](link)
 - MAY support:
   - [CreateMultipartUpload](link)